pkg/genericapiserver/genericapiserver.go

```diff
@@ -252,30 +252,26 @@ func (s *GenericAPIServer) Run(options *options.ServerRunOptions) {
 
 		}
 
-		glog.Infof("Serving securely on %s", secureLocation)
-		if options.TLSCertFile == "" && options.TLSPrivateKeyFile == "" {
-			options.TLSCertFile = path.Join(options.CertDirectory, "apiserver.crt")
-			options.TLSPrivateKeyFile = path.Join(options.CertDirectory, "apiserver.key")
-			// TODO (cjcullen): Is ClusterIP the right address to sign a cert with?
-			alternateIPs := []net.IP{s.ServiceReadWriteIP}
-			alternateDNS := []string{"kubernetes.default.svc", "kubernetes.default", "kubernetes", "localhost"}
-			// It would be nice to set a fqdn subject alt name, but only the kubelets know, the apiserver is clueless
-			// alternateDNS = append(alternateDNS, "kubernetes.default.svc.CLUSTER.DNS.NAME")
-			if !certutil.CanReadCertOrKey(options.TLSCertFile, options.TLSPrivateKeyFile) {
-				if err := certutil.GenerateSelfSignedCert(s.ClusterIP.String(), options.TLSCertFile, options.TLSPrivateKeyFile, alternateIPs, alternateDNS); err != nil {
-					glog.Errorf("Unable to generate self signed cert: %v", err)
-				} else {
-					glog.Infof("Using self-signed cert (%s, %s)", options.TLSCertFile, options.TLSPrivateKeyFile)
-				}
+		// TODO (cjcullen): Is ClusterIP the right address to sign a cert with?
+		alternateIPs := []net.IP{s.ServiceReadWriteIP}
+		alternateDNS := []string{"kubernetes.default.svc", "kubernetes.default", "kubernetes", "localhost"}
+		// It would be nice to set a fqdn subject alt name, but only the kubelets know, the apiserver is clueless
+		// alternateDNS = append(alternateDNS, "kubernetes.default.svc.CLUSTER.DNS.NAME")
+		if !certutil.CanReadCertOrKey(s.SecureServingInfo.ServerCert.CertFile, s.SecureServingInfo.ServerCert.KeyFile) {
```

Comment:
this changes the behavior to generate a self-signed cert if you typo the cert/key filename in the args... previously we only did that if they were unspecified